simphony/cuds/abstractparticles.py

```diff
@@ -17,48 +17,346 @@ class ABCParticles(object):
 
     @abstractmethod
     def add_particle(self, particle):
-        pass
+        """Adds the particle to the container.
+
+        If the new particle has no uid, the particle container
+        will generate a new uid for it. If the particle has
+        already an uid, it won't add the particle if a particle
+        with the same uid already exists. If the user wants to replace
+        an existing particle in the container there is an 'update_particle'
+        method for that purpose.
+
+        Parameters
+        ----------
+        particle : Particle
+            the new particle that will be included in the container.
+
+        Returns
+        -------
+        uid : uuid.UUID
+            The uid of the added particle.
+
+        Raises
+        ------
+        ValueError :
+            when the new particle already exists in the container.
+
+        Examples
+        --------
+        Having a Particle and a ParticleContainer just call the function
```

Comment:
`ParticleContainer`.   Should it be `Particles` or `particle container`?

(same question applies to docstring for `add_bond`)